src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,21 @@
+import PopupWithForm from "./PopupWithForm.js";
+
+export default class PopupWithConfirmation extends PopupWithForm {
+  constructor(popupSelector, handleFormSubmit, buttonText, loadingButtonText) {
+    super(popupSelector, handleFormSubmit, buttonText, loadingButtonText);
+  }
+
+  open(data) {
+    super.open();
+    this.data = data;
```

Comment:
There is no need to save the data, you can use it in index.js without passing to this class

Reply:
I'm having trouble understanding this one.

Reply:
If I don't save the data to this.data, then I can't pass it to inputValues. Sorry if I'm missing something.